test/wallet-rpc-test.js

```diff
@@ -0,0 +1,338 @@
+/* eslint-env mocha */
+
+'use strict';
+
+const {NodeClient,WalletClient} = require('bclient');
+const assert = require('./util/assert');
+const FullNode = require('../lib/node/fullnode');
+const Network = require('../lib/protocol/network');
+const Mnemonic = require('../lib/hd/mnemonic');
+const HDPrivateKey = require('../lib/hd/private');
+const KeyRing = require('../lib/primitives/keyring');
+const Script = require('../lib/script/script');
+const Address = require('../lib/primitives/address');
+const network = Network.get('regtest');
+
+// commonly used test mnemonic
+const phrase = [
+  'abandon', 'abandon', 'abandon', 'abandon',
+  'abandon', 'abandon', 'abandon', 'abandon',
+  'abandon', 'abandon', 'abandon', 'about'
+].join(' ');
+
+const node = new FullNode({
+  network: network.type,
+  apiKey: 'bar',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'bar'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'bar'
+});
+
+const watchOnlyWalletId = 'foo';
+const standardWalletId = 'bar';
+
+describe('Wallet RPC Methods', function() {
+  this.timeout(15000);
+
+  before(async () => {
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+  });
+
+  /*
+   * set up the testing environment
+   * by creating a wallet and a watch
+   * only wallet
+   */
+  it('setup: should start a fresh chain', async () => {
```

Comment:
Should the setup be a part of the `before` hook?

Reply:
Agreed and updated